core/includes/file.inc

```diff
@@ -2018,6 +2018,64 @@ function file_validate_image_orientation(File $file, $orientation = 0) {
   return $errors;
 }
 
+/**
+ * Validate uploaded SVG files.
+ *
+ * @param File $file
+ *   A file entity.
+ *
+ * @return
+ *   An array. If the file is an SVG and is either not valid or contains
+ *   dangerous content, it will contain an error message.
+ *
+ * @see hook_file_validate()
+ */
+function file_validate_svg(File $file) {
+  $errors = array();
+
+  if (image_is_svg($file->uri)) {
+    $file_contents = file_get_contents($file->uri);
+    $xml = simplexml_load_string($file_contents);
+
+    $errors = libxml_get_errors();
+
+    if (!$xml || !empty($errors)) {
+      return array(t('Invalid SVG file.'));
+    }
+    else {
+      $namespaces = $xml->getNamespaces();
+      if (!in_array('http://www.w3.org/2000/svg', $namespaces)) {
+        $errors[] = t('Invalid SVG namespace.');
```

Comment:
Hm... it's actually not invalid, but missing. That's not per se forbidden, as far as I read in the docs, but extremely uncommon. Needs more recherche. :thinking:

Reply:
Yeah, I had to think about this. It seems the namespace declaration is recommended, but not required? Maybe should remove this.

Reply:
Removing seems like an option, yes. Will test possible side effects.

Reply:
We could drop that check and make our xpath queries a little more greedy to still catch naughty content.
However, I can't get Firefox nor Chrome to display such svg *as image*. Both seem to be like "svg namespace or bust". :wink: Locally, in other software, those crippled svg's still render fine as image, but not in browsers. And browser rendering is what people need in a web CMS.

So I'd say let's stick with the namespace check - which validates that it's svg and not some arbitrary xml.

Reply:
Yep I found the same, lots of errors without that namespace, so keeping, agreed.